src/macaron/slsa_analyzer/build_tool/base_build_tool.py

```diff
@@ -184,3 +230,205 @@ def get_build_dirs(self, repo_path: str) -> Iterable[Path]:
 
         except StopIteration:
             pass
+
+    def serialize_to_json(self, cmd: list[str]) -> str:
+        """Convert a list of values to a json-encoded string so that it is easily parsable by later consumers.
+
+        Parameters
+        ----------
+        cmd: list[str]
+            List of command-line arguments.
+
+        Returns
+        -------
+        str
+            The list of command-line arguments as a json-encoded string.
+        """
+        return json.dumps(cmd)
+
+    def is_build_command(self, cmd: list[str]) -> bool:
+        """
+        Determine if the command is a build tool command.
+
+        Parameters
+        ----------
+        cmd: list[str]
+            List of command-line arguments.
+
+        Returns
+        -------
+        bool
+            True if the command is a build tool command.
+        """
+        # Check for empty or invalid commands.
+        if not cmd or not cmd[0]:
+            return False
+        # The first argument in a bash command is the program name.
+        # So first check that the program name is a supported build tool name.
+        # We need to handle cases where the first argument is a path to the program.
+        cmd_program_name = os.path.basename(cmd[0])
+        if not cmd_program_name:
+            logger.debug("Found invalid program name %s.", cmd[0])
+            return False
+
+        build_tools = set(itertools.chain(self.builder, self.packager, self.publisher, self.interpreter))
+        if any(tool for tool in build_tools if tool == cmd_program_name):
+            return True
+
+        return False
+
+    def match_cmd_args(self, cmd: list[str], tools: list[str], args: list[str]) -> bool:
+        """
+        Check if the build command matches any of the tools and the command-line arguments.
+
+        If build command's first element, which is the program name matches any of the `tools` names and any of its arguments
+        match any of the arguments in `args`, this function returns True.
+
+        If a build tool is called as a module, we parse the build command's args to extract the module name, then match
+        the subsequent arguments.
```

Comment:
It seems to me that the logic for handling this behavior is not located in this method.

Reply:
See commit [a882a62](https://github.com/oracle/macaron/pull/673/commits/a882a62dda353170d381273e9a1ca7fe235b03e0)